tests/std/tests/P2136R3_invoke_r/test.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <functional>
+#include <type_traits>
+
+using namespace std;
+
+// TRANSITION, DevCom-1457457
+namespace detail {
+    static constexpr bool permissive() {
+        return false;
+    }
+
+    template <class>
+    struct DependentBase {
+        static constexpr bool permissive() {
+            return true;
+        }
+    };
+
+    template <class T>
+    struct Derived : DependentBase<T> {
+        static constexpr bool test() {
+            return permissive();
+        }
+    };
+} // namespace detail
+constexpr bool is_permissive = detail::Derived<int>::test();
+
+constexpr int square(int n) {
+    return n * n;
+}
+constexpr int square_noexcept(int n) noexcept {
+    return n * n;
+}
+
+struct Thing {
+    int n = 0;
+    constexpr int& foo() {

```

Comment:
We conventionally avoid naming variables/functions/etc. `foo`, so this should be any other name (like `meow`, although it doesn't *have* to be a cat noise).

Reply:
Hopefully my single letter change is acceptable 😅.

Reply:
:cow:
===